cosmos/config.py

```diff
@@ -0,0 +1,122 @@
+"""Module that contains all Cosmos config classes."""
+
+from __future__ import annotations
+
+import shutil
+from dataclasses import dataclass, field
+from pathlib import Path
+from logging import getLogger
+
+from cosmos.constants import TestBehavior, ExecutionMode, LoadMode
+from cosmos.exceptions import CosmosValueError
+
+logger = getLogger(__name__)
+
+
+@dataclass
+class RenderConfig:
+    """
+    Class for setting general Cosmos config.
+
+    :param emit_datasets: If enabled test nodes emit Airflow Datasets for downstream cross-DAG
+    dependencies. Defaults to True
+    :param test_behavior: The behavior for running tests. Defaults to after each
+    :param load_method: The parsing method for loading the dbt model. Defaults to AUTOMATIC
+    :param select: A list of dbt select arguments (e.g. 'config.materialized:incremental')
+    :param exclude: A list of dbt exclude arguments (e.g. 'tag:nightly')
+    """
+
+    emit_datasets: bool = True
+    test_behavior: TestBehavior = TestBehavior.AFTER_EACH
+    load_method: LoadMode = LoadMode.AUTOMATIC
+    select: list[str] = field(default_factory=list)
+    exclude: list[str] = field(default_factory=list)
+
+
+@dataclass
+class ProjectConfig:
+    """
+    Class for setting project config.
+
+    :param dbt_project_path: The path to the dbt project directory. Example: /path/to/dbt/project
+    :param models_dir: The path to the dbt models directory within the project. Defaults to models
+    :param seeds_dir: The path to the dbt seeds directory within the project. Defaults to seeds
+    :param snapshots_dir: The path to the dbt snapshots directory within the project. Defaults to
+    snapshots
+    :param manifest: The path to the dbt manifest file. Defaults to None
+    """
+
+    dbt_project: str | Path
+    models: str | Path = "models"
+    seeds: str | Path = "seeds"
+    snapshots: str | Path = "snapshots"
+    manifest: str | Path | None = None
```

Comment:
It would be great if our attribute names make a distinction between what is a folder and what is a file path